services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/asn1/SrmDecoder.java

```diff
@@ -0,0 +1,123 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import us.dot.its.jpo.ode.api.models.messages.SrmDecodedMessage;
+import us.dot.its.jpo.ode.api.models.messages.DecodedMessage;
+import us.dot.its.jpo.ode.api.models.messages.EncodedMessage;
+import us.dot.its.jpo.ode.context.AppContext;
+import us.dot.its.jpo.ode.model.Asn1Encoding;
+import us.dot.its.jpo.ode.model.Asn1Encoding.EncodingRule;
+import us.dot.its.jpo.ode.model.OdeAsn1Data;
+import us.dot.its.jpo.ode.model.OdeAsn1Payload;
+import us.dot.its.jpo.ode.model.OdeSrmData;
+import us.dot.its.jpo.ode.model.OdeSrmMetadata;
+import us.dot.its.jpo.ode.model.OdeSrmPayload;
+import us.dot.its.jpo.ode.model.OdeData;
+import us.dot.its.jpo.ode.model.OdeHexByteArray;
+import us.dot.its.jpo.ode.model.OdeLogMetadata.RecordType;
+import us.dot.its.jpo.ode.model.OdeMsgPayload;
+import us.dot.its.jpo.ode.model.ReceivedMessageDetails;
+import us.dot.its.jpo.ode.model.RxSource;
+import us.dot.its.jpo.ode.plugin.j2735.builders.SRMBuilder;
+import us.dot.its.jpo.ode.util.JsonUtils;
+import us.dot.its.jpo.ode.util.XmlUtils;
+import us.dot.its.jpo.ode.util.XmlUtils.XmlUtilsException;
+
+
+@Component
+public class SrmDecoder implements Decoder {
+
+
+    @Override
+    public DecodedMessage decode(EncodedMessage message) {
+        
+        // Convert to Ode Data type and Add Metadata
+        OdeData data = getAsOdeData(message.getAsn1Message());
+
+        XmlUtils xmlUtils = new XmlUtils();
+
+        try {
+            // Convert to XML for ASN.1 Decoder
+            String xml = xmlUtils.toXml(data);
+
+            // Send String through ASN.1 Decoder to get Decoded XML Data
+            String decodedXml = DecoderManager.decodeXmlWithAcm(xml);
+            
+
+            // Convert to Ode Json 
+            OdeSrmData srm = getAsOdeJson(decodedXml);
+
+            // build output data structure
+            DecodedMessage decodedMessage = new SrmDecodedMessage(srm, message.getAsn1Message(), "");
+            return decodedMessage;
+            
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
+            return new SrmDecodedMessage(null, message.getAsn1Message(), e.getMessage());
+        } catch (Exception e) {
+            e.printStackTrace();
+            return new SrmDecodedMessage(null, message.getAsn1Message(), e.getMessage());
+        }
+    }
+
+    @Override
+    public OdeData getAsOdeData(String encodedData) {
+        OdeMsgPayload payload = new OdeAsn1Payload(new OdeHexByteArray(encodedData));
+
+        // construct metadata
+        OdeSrmMetadata metadata = new OdeSrmMetadata(payload);
+
+        //construct metadata
+        metadata = new OdeSrmMetadata(payload);
+        metadata.setOdeReceivedAt(DecoderManager.getCurrentIsoTimestamp());
+        metadata.setOriginIp(DecoderManager.getStaticUserOriginIp());
+        metadata.setRecordType(RecordType.srmTx);
```

Comment:
chore(blocking): only construct metadata once
```suggestion
        OdeSrmMetadata metadata = new OdeSrmMetadata(payload);
        metadata.setOdeReceivedAt(DecoderManager.getCurrentIsoTimestamp());
        metadata.setOriginIp(DecoderManager.getStaticUserOriginIp());
        metadata.setRecordType(RecordType.srmTx);
```